modules/src/mock_context.rs

```diff
@@ -27,31 +38,38 @@ pub struct MockContext {
     /// A list of `max_history_size` headers, ascending order by their height (latest is last).
     history: Vec<MockHeader>,
 
-    // All the connections in the store.
-    connections: HashMap<ConnectionId, ConnectionEnd>,
-
     /// The set of all clients, indexed by their id.
     clients: HashMap<ClientId, MockClientRecord>,
 
     // Association between client ids and connection ids.
     client_connections: HashMap<ClientId, ConnectionId>,
+
+    // All the connections in the store.
+    connections: HashMap<ConnectionId, ConnectionEnd>,
 }
 
 /// Returns a MockContext with bare minimum initialization: no clients, no connections are
 /// present, and the chain has Height(1). This should be used sparingly, mostly for testing the
 /// creation of new domain objects.
 impl Default for MockContext {
     fn default() -> Self {
-        Self::new(5, Height::new(0, 1))
+        Self::new(Height::new(0, 1))
     }
 }
 
+/// Implementation of internal interface for use in testing. The methods in this interface should
+/// _not_ be accessible to any ICS handler.
 impl MockContext {
-    pub fn new(max_history_size: usize, latest_height: Height) -> Self {
+    pub fn new(latest_height: Height) -> Self {
+        // A couple of predefined fields. Seems necessary to parametrize these so far.
+        let chain_id = ChainId::from_str("chainA-1").unwrap();
+        let max_history_size = 5;
```

Comment:
https://github.com/informalsystems/ibc-rs/pull/310/commits/93327ddc99705bdbdcd6be2476b1bed98cbd366e

Reply:
> Also it is possible to have a bad context after built, which actually happens for most of the tests now where the version_number of the chain_id is different than the mock header versions.

Indeed, this actually caused a bug in `advance_chain_height`. I originally fixed the inconsistency between version number and chain_id [here](https://github.com/informalsystems/ibc-rs/commit/8a8c5303b381d6ce9c297bc6024a50e5bbf1075c), but then I [removed the chain_id altogether](https://github.com/informalsystems/ibc-rs/commit/93327ddc99705bdbdcd6be2476b1bed98cbd366e), since it wasn't really necessary.

Reply:
> I liked it better with the `max_history_size` as parameter.

I reintroduced the parameter https://github.com/informalsystems/ibc-rs/pull/310/commits/bbad223a99103e8e0d4ed10968c58fcef872bbc4